### PR TITLE
Fix InterruptVectorDefault() for CH32V10x, CH32V20x and CH32V30x

### PR DESCRIPTION
**Pull request by hexaedron**
Fix for #424
New `InterruptVectorDefault()` definitions according to WCH's startup code examples.

### CODE REVIEW

File: ch32v003fun/ch32v003fun.c [hunk 1]
```diff
@@ -891,8 +897,8 @@ void DMA2_Channel3_IRQHandler( void ) 	__attribute__((section(".text.vector_hand
 void DMA2_Channel4_IRQHandler( void ) 	__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
 void DMA2_Channel5_IRQHandler( void ) 	__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
 void OTG_FS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
-void USBHSWakeup_IRQHandler( void )		__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
-void USBHS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
+void USBFSWakeup_IRQHandler( void )		__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
+void USBFS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
```
Comment (AlexanderMandera) [inline ch32v003fun/ch32v003fun.c:901]: Renaming `USBHS_IRQHandler` breaks the existing V30x USB HS example (see extralibs/hsusb_v30x.c).
For D8 (I think which is the no high-speed version), there is only a single USBFS_IRQHandler with no wakeup handler.
For D8C (like 305/307) there is both USBFS_IRQHandler, USBHS_IRQHandler and USBHSWakeup_IRQHandler.

File: ch32v003fun/ch32v003fun.c [hunk 2]
```diff
@@ -1119,121 +1129,329 @@ void InterruptVectorDefault()
...
+	.word   DMA2_Channel4_IRQHandler   /* DMA2 Channel 4 */ \n \
+	.word   DMA2_Channel5_IRQHandler   /* DMA2 Channel 5 */ \n \
+	.word   ETH_IRQHandler             /* ETH */ \n \
+	.word   ETHWakeUp_IRQHandler       /* ETH WakeUp */ \n \
+	.word   CAN2_TX_IRQHandler         /* CAN2 TX */ \n \
+	.word   CAN2_RX0_IRQHandler        /* CAN2 RX0 */ \n \
+	.word   CAN2_RX1_IRQHandler        /* CAN2 RX1 */ \n \
+	.word   CAN2_SCE_IRQHandler        /* CAN2 SCE */ \n \
+	.word   USBFS_IRQHandler           /* USBFS */ \n \
+	.word   USBFSWakeup_IRQHandler     /* USBHS Wakeup */ \n \
```
Comment (AlexanderMandera) [inline ch32v003fun/ch32v003fun.c:1430]:
```suggestion
	.word   USBHSWakeup_IRQHandler     /* USBHS Wakeup */ \n \
```

File: ch32v003fun/ch32v003fun.c [hunk 3]
```diff
@@ -1119,121 +1129,329 @@ void InterruptVectorDefault()
...
+	.word   USBFS_IRQHandler           /* USBHS */ \n \
```
Comment (AlexanderMandera) [inline ch32v003fun/ch32v003fun.c:1431]:
```suggestion
	.word   USBHS_IRQHandler           /* USBHS */ \n \
```

File: ch32v003fun/ch32v003fun.c [hunk 4]
```diff
@@ -891,8 +897,8 @@ void DMA2_Channel3_IRQHandler( void ) 	__attribute__((section(".text.vector_hand
 void DMA2_Channel4_IRQHandler( void ) 	__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
 void DMA2_Channel5_IRQHandler( void ) 	__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
 void OTG_FS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
-void USBHSWakeup_IRQHandler( void )		__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
-void USBHS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
+void USBFSWakeup_IRQHandler( void )		__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
```
Comment (AlexanderMandera) [inline ch32v003fun/ch32v003fun.c:900]:
```suggestion
void USBFS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
void USBHSWakeup_IRQHandler( void )		__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
```

File: ch32v003fun/ch32v003fun.c (hunk 1 above)
Comment (AlexanderMandera) [inline ch32v003fun/ch32v003fun.c:901]:
```suggestion
void USBHS_IRQHandler( void ) 			__attribute__((section(".text.vector_handler"))) __attribute((weak,alias("DefaultIRQHandler"))) __attribute__((used));
```

File: ch32v003fun/ch32v003fun.c [hunk 5]
```diff
@@ -1119,121 +1129,329 @@ void InterruptVectorDefault()
...
-"	.word   USBWakeUp_IRQHandler       /* 58: USB Wake up from suspend */ \n"
-#if defined(CH32V20x) || defined(CH32V30x)
 
-#if defined(CH32V30x)
-"   .word   TIM8_BRK_IRQHandler        /* 59: TIM8 Break == masked to USBHD? */ \n"
-#else
-"	.word   USBHD_IRQHandler           /* 59: USBHD Break */ \n"
+#if defined(CH32V10x)
+"	.word   SysTick_Handler            /* SysTick Handler */ \n \
+    	.word   0 \n \
```
Comment (AlexanderMandera) [inline ch32v003fun/ch32v003fun.c:1135]: Removing the additional tab indent here should fix some of the unnecessary diff changes.

**Review by AlexanderMandera — CHANGES_REQUESTED** [review]

**Comment by cnlohr** [comment]
Merging but I will go back later and algebra out the interrupt table.

**Pull request merged**